app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers.php

```diff
@@ -578,6 +422,311 @@ protected function addError($errorMessage)
+                    }
+                }
+                break;
+            case 'billing_country':
+            case 'shipping_country':
+                $address = $this->getAddressForCustomizedAttributes($customAtt, $customer, $lastOrder);
+
+                if ($address) {
+                    $countryCode = $address->getCountry();
+                    if ($countryCode) {
+                        $countryName = Mage::getModel('directory/country')->loadByCode($countryCode)->getName();
+                        $mergeVars[$key] = $countryName;
+                    }
+                }
+                break;
+            case 'billing_zipcode':
+            case 'shipping_zipcode':
+                $address = $this->getAddressForCustomizedAttributes($customAtt, $customer, $lastOrder);
+
+                if ($address) {
+                    $zipCode = $address->getPostcode();
+                    if ($zipCode) {
+                        $mergeVars[$key] = $zipCode;
+                    }
+                }
+                break;
+            case 'billing_state':
+            case 'shipping_state':
+                $address = $this->getAddressForCustomizedAttributes($customAtt, $customer, $lastOrder);
+
+                if ($address) {
+                    $state = $address->getRegion();
+                    if ($state) {
+                        $mergeVars[$key] = $state;
+                    }
+                }
+                break;
+            case 'dop':
+                $dop = $helper->getLastDateOfPurchase($subscriberEmail, $lastOrder);
+                if ($dop) {
+                    $mergeVars[$key] = $dop;
+                }
+                break;
+            case 'store_code':
+                $storeCode = Mage::getModel('core/store')->load($storeId)->getCode();
+                $mergeVars[$key] = $storeCode;
+        }
+        Mage::log($mergeVars[$key], null, 'testing.log', true);
+
+        return $mergeVars;
+    }
+
+    /**
+     * @param $subscriber
+     * @param $attributeCode
+     * @param $customer
+     * @param $lastOrder
+     * @param $mergeVars
+     * @param $key
+     * @param $storeId
+     * @param $attribute
+     * @return array
+     */
+    protected function customerAttributes($subscriber, $attributeCode, $customer, $lastOrder, $mergeVars, $key, $storeId, $attribute)
+    {
+        switch ($attributeCode) {
+            case 'email':
+                break;
+            case 'default_billing':
+            case 'default_shipping':
+                $address = $customer->getPrimaryAddress($attributeCode);
+                $addressData = $this->getAddressData($address);
+
+                $addressData = $this->getAddressFromLastOrder($lastOrder, $addressData);
+
+                if (count($addressData)) {
+                    $mergeVars[$key] = $addressData;
+                }
+                break;
+            case 'gender':
+                if ($customer->getData($attributeCode)) {
```

Comment:
Extract to function getCustomerGroupLabel to adapt to the same format as other cases